### ISSUE TITLE
MSL question: can mem_flags be combined in threadgroup_barrier?

### ISSUE THREAD

**Issue by dneto0** [gpuweb/gpuweb#1547]
tl;dr:      can I write  `        threadgroup_barrier(mem_device |  mem_threadgroup)` in MSL?

The MSL threadgroup barrier function prototype is:

       void threadgroup_barrier(mem_flags flags);

Elsewhere, mem_flags is described as a bit field.
The examples in the spec only show a single bit being used at a time, e.g for mem_device, mem_threadgroup.

It's not clear if I can write:

        threadgroup_barrier(mem_device |  mem_threadgroup)

That is exactly the analog of what you'd have to write in OpenCL C 2.0 in order to make common idioms work:
Correct use of OpenCL 2.0 (that has its own memory model), usually requires that you combine the local and global mem fence flags in order to get the ordering most programmers expect. That's because the global-happens-before and local-happens-before orderings are *separately defined*, and only join when you do a memory barrier with *both* bits.

An alternate pattern is what @alan-baker hinted at in a WGSL meeting a few weeks ago:  have MSL support chaining a happens-before via barrier-to-barrier relation.  It's subtle.

**Comment by Kangz** [comment]
Is this still current? Barriers have been added to WGSL and don't use flags.
